Configuration/Config.cs

```diff
@@ -83,13 +83,27 @@ private static void Initialize()
         }
         
         /// <summary>
+        /// Sets a custom file name
+        /// </summary>
+        /// <param name="customFileName"></param>
+        private static void SetCustomFileName(string customFileName)
+        {
+            _loaded = false;
```

Comment:
This is not thread-safe at all. Imagine the case where thread 1 comes in and requests the normal config, but is then suspended, thread 2 comes in for a custom file and sets the _config static field and is then suspended. Thread 1 resumes and reads value from the new value that thread 2 set.